crates/wasi-common/wig/src/wasi.rs

```diff
@@ -291,6 +299,14 @@ pub fn define_struct_for_wiggle(args: TokenStream) -> TokenStream {
             linker_add.push(quote! {
                 linker.define(#module_name, #name, self.#name_ident.clone())?;
             });
+            // `proc_exit` is special; it's essentially an unwinding primitive,
+            // so we implement it in the runtime rather than in wasi-common.
+            if name == "proc_exit" {
+                ctor_externs.push(quote! {
+                    let #name_ident = wasmtime::Func::exit_func(store);
```

Comment:
Even though we're not exposing the `wig` generated snapshot 0 code to users over in the lucet embedding, this would make us take a dep on `wasmtime` in `lucet_runtime`, which would not be good.

Reply:
This code already depends on `wasmtime::Func::wrap`; how does that work in Lucet?

Reply:
I missed that in my first look, any dep inside `wasi-common` on `wasmtime` (or anything besides wig/wiggle) would be a showstopper for integration with Lucet. (I'm on vacation this week and will not be around much, just popped in from a lack of discipline)

Reply:
@kubkon `define_struct` has the comment "This is a single-use macro intended to be used in the `wasmtime-wasi` crate.". Do you know if that's intended to be the case for `define_struct_for_wiggle`? `define_struct_for_wiggle` currently has Wasmtime-specific code in it, so if it's meant to be usable outside of Wasmtime, we'll need to figure out how to factor that out.

Reply:
I’m pretty sure `wig` is fully internal to `Wasmtime`, and so is `define_struct_for_wiggle`. As far as I remember, `lucet` had their own set of magical macros, but @pchickey can correct me here if I’m wrong :-)